### DISCUSSION TITLE
Example EEBus device implementation with usecases

### DISCUSSION THREAD

**Discussion by Nibot1** [enbility/eebus-go#64] [Q&A]
Ist there any working example for an EEBUS device/client implementation with implemented usecases like a wallbox? I would like to simulate an DHWStorgage to my Viessmann Gridbox with an Vitocharge VX3 to turn on a relay to heat the DHW-Heatpump, because ther is no SG-Ready Relay in the Inverter.

**Comment by DerAndereAndi** [comment]
Hi,

any service can provide client and server features and does so. Currently I only implemented usecases from a CEM perspective in https://github.com/enbility/cemd and I don't know of anyone doing so for a different device type yet. But in general it should be similarly possible.

Every usecase spec covers both ends and the difference basically only is, which one requests data, and which one provides them for which part. There might be some helpers missing for doing things easier, but in general there should be no showstopper.

I am not sure if the Gridbox already supports the DHWStorage usecases. At least they do not show up in the supported usecases when connecting to a VitoConnect. See https://enbility.net/devices/
